docs/user/filesystem.md

```diff
@@ -1021,6 +1021,14 @@ libpil4dfs intercepting summary for ops on DFS:
 [op_sum ]  5003
 ```
 
+### Turn on whitelist mode in libpil4dfs
+libpil4dfs could boost IO performance considerably over using dfuse for IO intensive workloads. However, initializing daos environment in libpil4dfs is expensive. To alleviate the overhead of libpil4dfs, users could set env to disable function interception in specific applications. I.e., let dfuse handle all IO requests. "D_IL_WHITELIST=1" turns on whitelist mode. A list of common applications are hardcoded to be skipped in whitelist mode. Users can set env "D_IL_BYPASS_ALL_LIST" to bypass function interceptions in target processes and their child processes. "D_IL_BYPASS_LIST" can be set to bypass function interceptions in target processes only. Setting "D_IL_BYPASS_ALL_LIST" or "D_IL_BYPASS_LIST" enables whitelist mode too even "D_IL_WHITELIST=1" is not set. Whitelist mode is recommended for short-lived processes that run a large number (more than 1,000) of times. Example to set env:
```

Comment:
i thought we want the whitelist mode ON by default for such applications?
it's always handy to add env variables but most of those are going to be for dev use only and in production i bet no one would use them.
The other comment here is that we need to use some other term than whitelist. it's too vague for use here.

as for the other 2 env variables, im not sure why there is any use case where we want to have them separated. again if you think this is valuable for development, let's just hide 1 and expose just 1 of those.

Another thing that is not clear from this description. does the bypass list override the whitlist applications? or does it append to them?

Reply:
My only concern to set whitelist mode as default is that the applications in the built-in bypass list will not be fully tested with libpil4dfs. The bugs in libpil4dfs will not be exposed in those applications. We probably will make he built-in bypass list shorter if we decide to make whitelist mode as default.

You are right. Most regular users do not read user manual carefully and they may be not aware of available env variables. Maybe we can add some description about compatible mode and whitelist mode in frequently asked question section. In case libpil4dfs breaks an application, users can compatible mode and whitelist mode.

> The other comment here is that we need to use some other term than whitelist. it's too vague for use here.
ok. I will rephrase the words here to be more clear.

> as for the other 2 env variables, im not sure why there is any use case where we want to have them separated. again if you think this is valuable for development, let's just hide 1 and expose just 1 of those.

You are right. We can expose only one env variable "D_IL_BYPASS_ALL_LIST" to make it simpler.

> Another thing that is not clear from this description. does the bypass list override the whitlist applications? or does it append to them?
The built-in bypass list can not be overridden. The user provided will be appended. I will make it more clear in document. Thank you!

Reply:
> My only concern to set whitelist mode as default is that the applications in the built-in bypass list will not be fully tested with libpil4dfs. The bugs in libpil4dfs will not be exposed in those applications. We probably will make he built-in bypass list shorter if we decide to make whitelist mode as default.
> 
> You are right. Most regular users do not read user manual carefully and they may be not aware of available env variables. Maybe we can add some description about compatible mode and whitelist mode in frequently asked question section. In case libpil4dfs breaks an application, users can compatible mode and whitelist mode.

im going to push back on this.
what is the purpose of this PR? is whitelisting to avoid interception recommended or not? if yes, then this should be the default, if not then this PR should really not be merged or it can be a dev only thing.
No matter how much you publicize env variables, they are not a good option for production. no one is using them unless the admin makes it inherit in user environments, which is the same really as making it default in the first place.

if your concern is testing, then why not run testing with the env variable set (to turn that off)?
in production your goal should be to make things faster and avoid hitting bugs / issues and not finding bugs. the goal for regression and system tests is to find bugs.

Reply:
I agree. I will enable whitelist mode as default and disable it in our testing in CI. Thank you very much!